### ISSUE TITLE
Ability to reference SDK without the version

### ISSUE THREAD

**Issue by martincostello** [microsoft/testfx#2703]
## Summary

_I don't know if this a general MSBuild restriction, or specific to the new SDK, so this might not be the right place to feedback on this._

Add the ability to reference the SDK without the version number, for example:

```diff
- <Project Sdk="MSTest.Sdk/3.3.1">
+ <Project Sdk="MSTest.Sdk">
```

## Background and Motivation

Requiring the `Sdk` to include the version is not friendly to package updating tools or NuGet Central Package Version Management where the aim is to define the package version(s) in exactly one place and to then allow them to be easily maintained in an automated version with tools such as dependabot or dotnet-outdated.

I experimented with using an MSBuild property for the version, but MSBuild does not appear to use the value late enough in project loading to be able to use that approach.

## Proposed Feature

_Somehow_ allow the SDK to be loaded/referenced without an explicit version number by loading it from the package reference (e.g. from `<PackageVersion Include="MSTest.TestFramework" Version="3.3.1" />`).

## Alternative Designs

None.

**Comment by MarcoRossignoli** [comment]
> Somehow allow the SDK to be loaded/referenced without an explicit version number by loading it from the package reference (e.g. from <PackageVersion Include="MSTest.TestFramework" Version="3.3.1" />).

Hi @martincostello, the MSBuild sdk versioning doesn't allow the usage of the `NuGet Central Package Version Management`

As you can see here https://github.com/microsoft/testfx/tree/main/src/Package/MSTest.Sdk/Sdk an MSBuild SDK is a bunch of props/targets that we compose  (for instance this is the main file https://github.com/microsoft/testfx/blob/main/src/Package/MSTest.Sdk/Sdk/Runner/ClassicEngine.targets).

So it's not easy have 1 version for everything because the imports will import more packages with different versions. So our 3.1.1 sdk actually is the "version" of MSTest from feature perspective but it's composed by more packages from different part of the platform and with different versions. For instance the new testing platform(https://github.com/microsoft/testfx/blob/main/docs/testingplatform/Index.md) that empower the runner is at the moment at version 1.1.0.

The only way at the moment to centralize the version at solution level is to use the `global.json` as explained here https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk?view=vs-2022#how-project-sdks-are-resolved

**Comment by martincostello** [comment]
Thanks for the information - I'll have a look to see if adding the SDK in `global.json` works.

**Comment by martincostello** [comment]
That did the trick - maybe worth calling out in some documentation somewhere? I originally just copy-pasted from the samples in the blog post.

**Comment by MarcoRossignoli** [comment]
> That did the trick - maybe worth calling out in some documentation somewhere? I originally just copy-pasted from the samples in the blog post.

Good idea, it's in the note box under the code sample https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk but looks like is not super clear, thanks for the suggestion.

**Comment by Evangelink** [comment]
TODOs:

- Update doc to make it clearer that `global.json` can be used and provide a full example
- Make it clear this is the recommended pattern
- Update SDK sample to add the global.json case
